master/buildbot/test/unit/test_secret_in_file.py

```diff
@@ -0,0 +1,114 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+import stat
+
+from twisted.internet import defer
+from twisted.python.filepath import FilePath
+from twisted.trial import unittest
+
+from buildbot.secrets.providers.file import SecretInAFile
+from buildbot.test.util.config import ConfigErrorsMixin
+
+
+class TestSecretInFile(ConfigErrorsMixin, unittest.TestCase):
+
+    def createTempDir(self, dirname):
+        tempdir = FilePath(self.mktemp())
+        tempdir.createDirectory()
+        return tempdir.path
+
+    def createFileTemp(self, tempdir, filename, text=""):
+        file_path = os.path.join(tempdir, filename)
+        with open(file_path, 'w') as filetmp:
+            filetmp.write(text)
+        return filetmp, file_path
+
+    @defer.inlineCallbacks
+    def setUp(self):
+        self.tmp_dir = self.createTempDir("temp")
+        filetmp, self.filepath = self.createFileTemp(self.tmp_dir,
+                                                     "tempfile.txt",
+                                                     text="key value")
+        os.chmod(self.filepath, 0o700)
+        self.srvfile = SecretInAFile(self.tmp_dir)
+        yield self.srvfile.startService()
+
+    @defer.inlineCallbacks
+    def tearDown(self):
+        yield self.srvfile.stopService()
+        # windows requires to close all existing open temp files not closed by
```

Comment:
looks dirty. Why open temp files were not open with "with open" in the first place?